src/hotspot/share/runtime/globals.hpp

```diff
@@ -1362,6 +1362,21 @@ const int ObjectAlignmentInBytes = 8;
           "-XX:MallocLimit=2g:oom"                                          \
           "-XX:MallocLimit=compiler:200m:oom,code:100m")                    \
                                                                             \
+                                                                            \
+  product(ccstr, RssLimit, nullptr, DIAGNOSTIC,                             \
+          "Limit to Resident Set Size. The JVM will periodically "          \
+          "check if that limit had been reached and, if true, "             \
+          "generate a fatal error. The limit can be either given as a "     \
+          "memory size (e.g. \"2G\") or as a percentage of "                \
+          "the total available memory on this machine or in this "          \
+          "container (e.g. \"-XX:RssLimit=80%%\"). A value of 0 (default) " \
```

Comment:
It would be more usual to take this as a fraction of available memory e.g. 0.8.

That simplifies the parsing and validation logic.

Reply:
I think percent is easier to understand:

- RSSLimit=100  - percent, I guess? Because such a small number makes no sense as limit?
- RSSLimit=90 -  percent, I guess?
- RSSLimit=90000000 - not percent, because it is larger than 100?
- RSSLimit=100.0 - percent because of the decimal point?

Alternative would be a complementary "RssLimitPercent" switch, but I want to keep the number of switches at a minimum.

Reply:
It would be 0.9 == 90% etc. so 0.x to 1.0